### PR TITLE
Define backward-compatible attached clusters API for the direct controller

### PR DESCRIPTION
**Pull request by hankfreund**
### Change description
- Generated the configconnector types for ContainerAttachedCluster
- Manually updated the types to match the existing API
- Regenerated the CRD to ensure no breaking changes

### Tests you have done
- Manually ran all the attached cluster mockGCP tests

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [x] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_containerattachedclusters.containerattached.cnrm.cloud.google.com.yaml [hunk 1]
```diff
@@ -222,40 +256,48 @@ spec:
...
+                x-kubernetes-validations:
+                - message: ResourceID field is immutable
+                  rule: self == oldSelf
             required:
             - distribution
             - fleet
             - location
             - oidcConfig
             - platformVersion
-            - projectRef
```
Comment (yuwenma) [inline config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_containerattachedclusters.containerattached.cnrm.cloud.google.com.yaml:245]: Is project not required?

**Review by yuwenma — COMMENTED** [review]
/approve

One question about the optional/required field. Otherwise looks good.

File: config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_containerattachedclusters.containerattached.cnrm.cloud.google.com.yaml [hunk 2]
```diff
@@ -43,79 +44,93 @@ spec:
...
+                  Immutable. The Kubernetes distribution of the underlying attached cluster.
+
+                   Supported values: ["eks", "aks", "generic"].
                 type: string
+                x-kubernetes-validations:
+                - message: Distribution field is immutable
+                  rule: self == oldSelf
               fleet:
-                description: Fleet configuration.
+                description: Required. Fleet configuration.
```
[the remark this answers is not in the document]
Comment (hankfreund) [reply config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_containerattachedclusters.containerattached.cnrm.cloud.google.com.yaml:122]: This has always been required by the multicloud API, so if CC didn't require it, that was a bug. 

The public documentation for CC here https://cloud.google.com/config-connector/docs/reference/resource-docs/containerattached/containerattachedcluster indicates the field is required, so this shouldn't be a breaking change regardless.

File: config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_containerattachedclusters.containerattached.cnrm.cloud.google.com.yaml (hunk 1 above)
Comment (hankfreund) [reply config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_containerattachedclusters.containerattached.cnrm.cloud.google.com.yaml:245]: I guess this is a little confusing. A project is required, but according to the docs: "If it is not provided, the provider project is used", which seems like it's optional?

But it _was_ required prior, so I'll put it back.

**Comment by yuwenma** [comment]
/lgtm
/approve 

Thanks for the clarification. PR looks good! Could you fix the presubmit linter error ?

**Pull request merged**